includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -2048,6 +2052,37 @@ private function check_attr_spec_rule_disallowed_empty( DOMElement $node, $attr_
 		return AMP_Rule_Spec::NOT_APPLICABLE;
 	}
 
+	/**
+	 * Check if attribute has multiple image candidates with the same width or pixel density.
+	 *
+	 * @since 2.0.2
+	 *
+	 * @param DOMElement $node      Node.
+	 * @param string     $attr_name Attribute name.
+	 *
+	 * @return int:
+	 *      - AMP_Rule_Spec::PASS - $attr_name has a value that matches the rule.
+	 *      - AMP_Rule_Spec::FAIL - $attr_name has a value that does *not* match rule.
+	 *      - AMP_Rule_Spec::NOT_APPLICABLE - $attr_name does not exist or there are no duplicate dimensions.
+	 */
+	private function check_attr_spec_rule_duplicate_dimension( DOMElement $node, $attr_name ) {
+		if (
+			Attribute::SRCSET === $attr_name
+			&&
+			$node->hasAttribute( $attr_name )
+			&&
+			preg_match_all( '/(?:[^,\s]\S*[^,\s])\s*([\d]+.?[\d]*[wx])?\s*(?:(,)\s*)?/', $node->getAttribute( $attr_name ), $matches )
```

Comment:
See note above about perhaps needing to use `preg_split()` to be more resilient.